riscv/src/continuations/bootloader.rs

```diff
@@ -32,6 +37,225 @@ pub const MEMORY_HASH_START_INDEX: usize = 2 * REGISTER_NAMES.len();
 pub const NUM_PAGES_INDEX: usize = MEMORY_HASH_START_INDEX + WORDS_PER_HASH * 2;
 pub const PAGE_INPUTS_OFFSET: usize = NUM_PAGES_INDEX + 1;
 
+/// This trait provides all the field specific types and implementations that the bootloader needs.
+///
+/// For now, this trait is implemented directly by each `FieldElement` type, because the hash functions (i.e., poseidon) are field-specific.
+pub(crate) trait BootloaderImpl<F> {
```

Comment:
main reason for this trait here are the `Page` and `Hash` types, which are just arrays of `F` but with different sizes for the small/large machines.
These are used in the merkle tree and bootloader code, and this is the way I found to make it generic.